harbor-ui/src/routes/receive.rs

```diff
@@ -25,26 +26,50 @@ pub fn receive(harbor: &HarborWallet) -> Element<Message> {
                 h_header("Deposit", "Receive on-chain or via lightning.")
             };
 
+            let generating = harbor.receive_status == ReceiveStatus::Generating;
+
             let amount_input = h_input(InputArgs {
                 label: "Amount",
                 placeholder: "420",
                 value: &harbor.receive_amount_str,
                 on_input: Message::ReceiveAmountChanged,
                 numeric: true,
                 suffix: Some("sats"),
+                disabled: generating,
                 ..InputArgs::default()
             });
 
-            let generating = harbor.receive_status == ReceiveStatus::Generating;
+            let generate_invoice_button = || {
+                h_button("Generate Invoice", SvgIcon::Qr, generating)
+                    .on_press(Message::GenerateInvoice)
+            };
 
-            let generate_button = h_button("Generate Invoice", SvgIcon::Qr, generating)
-                .on_press(Message::GenerateInvoice);
+            let generate_address_button = || {
+                h_button("Generate Address", SvgIcon::Qr, generating)
+                    .on_press(Message::GenerateAddress)
+            };
 
-            let generate_address_button = h_button("Generate Address", SvgIcon::Qr, generating)
-                .on_press(Message::GenerateAddress);
+            let start_over_button = || {
+                h_button("Start Over", SvgIcon::Restart, false)
+                    .on_press(Message::CancelReceiveGeneration)
+            };
```

Comment:
why are these functions now, can you not just clone?

Reply:
you can't clone widgets... this led me down a dark path where I rewrote receive so it was less nested / easier to reason about